ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java

```diff
@@ -5611,4 +5365,282 @@ private enum TableType {
     JDBC
   }
 
+  private class OrderByRelBuilder {
```

Comment:
Since a lot of code has been refactored and changed place as part of this change it may be a good opportunity to extract this refactored class completely outside of the `CalcitePlanner` if that is possible (keeping it package private).

Reply:
Moved this class to its own file.

Reply:
I reverted this change because it caused a Circular dependency which according to Sonar it makes maintenance harder.